classes/cinema_ticket_processor_class.rb

```diff
@@ -0,0 +1,24 @@
+class CinemaTicketsProcessor
+  attr_accessor :movie_name, :ticket_cost, :screening_time
+
+  def initialize(movie_name, ticket_cost, screening_time)
+    @movie_name = movie_name
+    @ticket_cost = ticket_cost
+    @screening_time = screening_time
+  end
+
+  def ticket_printer
+    puts @movie_name
+    puts @screening_time
+    puts @ticket_cost
+    puts seat_selection
+  end
+
+  private
+    def seat_selection
+      rand 1..150
+    end
+end
+
+cinema_ticket = CinemaTicketsProcessor.new('Titanic', '20 zł', '18:00')
+cinema_ticket.ticket_printer
```

Comment:
`CinemaTicketsProcessor`

Usually the name like `processor`, `handler`, `coordinator`, `manager` is an indicator that the class responsibility is too generic. Also, it's like an an invitation to put any kind of logic into this class in the future ("processing" means almost anything, so everything fits into that class :)).

It is not a hard rule that you can not use that name. I'm only saying that it is some signal that should cause you to think for a moment about whether you are choosing the best name or not.

For example, in your case, you could make it more specific: `CinemaTicketsPrinter`. In this way you make it clear what's the responsibility of this class and you increase the chances that the class will have a "single responsibility" in the future.

----

But in your case, I've recommend, calling it `Ticket`:

```ruby
class Ticket
  # ....
end
```  

Because looking at the interface:

```ruby
Something.new('Titanic', '20 zł', '18:00')
```

I think you want to represent "the ticket":

```ruby
Ticket.new('Titanic', '20 zł', '18:00')
```

Reply:
After you change the class name to `Ticket`, you could think a little bit more about the public interface of this class:

Imagine how it will be used by someone from the outside:

```ruby
ticket = Ticket.new('Titanic', '20 zł', '18:00')
ticket.ticket_printer
```

What do you think about this interface? Personally, I'd change 3 things, but I will let you think about it yourself. Maybe you will decide that this is the best interface and that's fine as well. But think about it consciously :)